lang_js/parsing/ast_js.ml

```diff
@@ -351,6 +351,11 @@ and type_ =
 and type_parameter = ident (* TODO: constraints *)
 and type_parameter_constraint = type_
 
+and tuple_type_member =
+  | TyTupMember of type_  (* simple tuple type element *)
+  | TyTupOpt of type_  (* optional tuple type element in typescript: string? *)
```

Comment:
What's wrong with using TyQuestion for that?

Reply:
TyQuestion refers to the syntax while TyTupOpt refers to the semantics, which could exist in other languages under another syntax. For the common case, I went with TyTupMember but logically it should be TyTupRequired. :shrug:

Reply:
I'm ok to be be as specifc as possible in the language-specifc AST, but for AST generic I prefer to be more general.